test/util/kubernetes.go

```diff
@@ -71,40 +71,43 @@ func GetPods(cl kubernetes.Interface, ns string) []string {
 
 // GetAppPods awaits till all pods are running in a namespace, and returns a map
 // from "app" label value to the pod names.
-func GetAppPods(cl kubernetes.Interface, ns string) (map[string][]string, error) {
+func GetAppPods(cl kubernetes.Interface, nslist []string) (map[string][]string, error) {
 	pods := make(map[string][]string)
 	var items []v1.Pod
-	for n := 0; ; n++ {
-		glog.Info("Checking all pods are running...")
-		list, err := cl.CoreV1().Pods(ns).List(meta_v1.ListOptions{})
-		if err != nil {
-			return pods, err
-		}
-		items = list.Items
-		ready := true
 
-		for _, pod := range items {
-			if pod.Status.Phase != "Running" {
-				glog.Infof("Pod %s has status %s", pod.Name, pod.Status.Phase)
-				ready = false
-				break
-			}
-		}
+	for _, ns := range nslist {
+		glog.Infof("Checking all pods are running in namespace %s ...", ns)
 
-		if ready {
-			break
-		}
+		for n := 0; ; n++ {
+			list, err := cl.CoreV1().Pods(ns).List(meta_v1.ListOptions{})
+			if err != nil {
+				return pods, err
+			}
+			items = list.Items
+			ready := true
+
+			for _, pod := range items {
+				if pod.Status.Phase != "Running" {
```

Comment:
I seem to recall adding something along the lines of checking every container's status in the pod, as opposed to just the pod status (because k8s says pod is running even if only one container is running).